engine/framework/worker.go

```diff
@@ -233,14 +233,12 @@ func (w *DefaultBaseWorker) Init(ctx context.Context) error {
 
 // NotifyExit implements BaseWorker.NotifyExit
 func (w *DefaultBaseWorker) NotifyExit(ctx context.Context, errIn error) (retErr error) {
-	w.closeImplOnce.Do(func() {
-		// Must ensure that the business logic is
-		// notified before closing.
-		w.callCloseImpl()
-	})
+	w.partialClose()
 
 	startTime := time.Now()
 	defer func() {
+		w.cleanMessageHandler()
+
```

Comment:
We have no semantic restriction on a bunch of `clean` functions, like `cleanMessageHandler` and `doClose`.
It's dangerous to call these functions without any protection like `sync.Once`.